power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -33,21 +33,35 @@ template <component_c Component, class ComponentContainer,
           std::output_iterator<Idx2D> OutputIterator>
     requires model_component_state_c<MainModelState, ComponentContainer, Component>
 inline void get_component_sequence(MainModelState<ComponentContainer> const& state, ForwardIterator begin,
-                                   ForwardIterator end, OutputIterator destination) {
+                                   ForwardIterator end, OutputIterator destination, Idx n_comp_elements) {
     using UpdateType = typename Component::UpdateType;
 
-    std::transform(begin, end, destination,
-                   [&state](UpdateType const& update) { return get_component_idx_by_id<Component>(state, update.id); });
+    if (n_comp_elements == na_Idx) {
+        auto idx_getter_default = [&state](UpdateType const& update) {
+            return get_component_idx_by_id<Component>(state, update.id);
+        };
+        std::ranges::transform(begin, end, destination,
+                               [&](UpdateType const& update) { return idx_getter_default(update); });
+    } else {
+        auto idx_getter_func = [&state](auto index) {
+            Idx const group = get_component_group_idx<Component>(state);
+            return Idx2D{group, index};
+        };
+        std::ranges::transform(begin, end, destination, [&, index = 0](UpdateType const& /*update*/) mutable {
```

Comment:
since `UpdateType` is not used, if you use `auto`, it may be able to work both with `Proxy` and `UpdateType` with zero-copy. currently, it would use a copy for `Proxy` (due to cast to `UpdateType`